openedx/features/pakx_features/views.py

```diff
@@ -13,20 +14,61 @@
     sort_by_announcement,
     sort_by_start_date
 )
-from django.views.decorators.csrf import ensure_csrf_cookie
-from common.djangoapps.util.cache import cache, cache_if_anonymous
+from student.models import CourseEnrollment
 from edxmako.shortcuts import render_to_response
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.csrf import ensure_csrf_cookie
 from openedx.core.djangoapps.catalog.utils import get_programs, get_programs_with_type
 from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.course_experience.utils import (get_course_outline_block_tree,
+                                                      get_resume_block)
 # Create your views here.
 
 
+def add_resume_link_for_course(request, course, allow_start_dates_in_future=False):
+    """
+    Returns information relevant to resume course functionality.
+
+    Returns a tuple: (has_visited_course, resume_course_url, resume_course_title)
+        has_visited_course: True if the user has ever visited the course, False otherwise.
+        resume_course_url: The URL of the 'resume course' block if the user has visited the course,
+            otherwise the URL of the course root.
+        resume_course_title: The display_name of the resume course block, otherwise the display_name of course root
+
+    """
+    course_id = text_type(course.id)
+    course_outline_root_block = get_course_outline_block_tree(request, course_id, request.user,
+                                                              allow_start_dates_in_future)
+    resume_block = get_resume_block(course_outline_root_block) if course_outline_root_block else None
+    has_visited_course = bool(resume_block)
+    if resume_block:
+        resume_course_url = resume_block['lms_web_url']
+        resume_course_title = resume_block['display_name']
+    else:
+        resume_course_url = course_outline_root_block['lms_web_url'] if course_outline_root_block else None
+        resume_course_title = course_outline_root_block['display_name'] if course_outline_root_block else None
```

Comment:
```suggestion
     resume_course_url = None
     resume_course_title = None
     if ...
          ...
    elif course_outline_root_block:
        resume_course_url = course_outline_root_block['lms_web_url']
        resume_course_title = course_outline_root_block['display_name']
```

Reply:
These values will be assigned `none` when `else` case is executed.